lib/plugin.h

```diff
@@ -67,11 +67,14 @@ struct _Plugin
   const gchar *name;
   PluginFailureInfo failure_info;
   CfgParser *parser;
-  void (*setup_context)(Plugin *self, GlobalConfig *cfg, gint plugin_type, const gchar *plugin_name);
-  gpointer (*construct)(Plugin *self, GlobalConfig *cfg, gint plugin_type, const gchar *plugin_name);
+  gpointer (*construct)(Plugin *self);
   void (*free_fn)(Plugin *s);
 };
 
+gpointer plugin_construct(Plugin *self);
+gpointer plugin_construct_from_config(Plugin *self, CfgLexer *lexer, gpointer arg);
+
+typedef struct _ModuleInfo ModuleInfo;
```

Comment:
Sorry for bothering you again, but this is also a redefinition. The first one is at line 31.

Reply:
I have now removed these. I think they got here because a bogus conflict resolution.

btw: I have now tried to compile gcc-4.7 and it still compiles the multiple typedefs. How far do you need to go in gcc archeology to not support this?

I have even tried with -std=c99 but that didn't compile ivykis, gnu90 does compile the typedefs.

anyway they shouldn't be there, so the gcc version is probably not that important

Reply:
We tried to compile this with gcc (Ubuntu 4.4.3-4ubuntu5) 4.4.3 (it is the linux-glibc2.11 builder).

This version of gcc fails without using any flags (no pedantic or std=c99 needed).

Reply:
```
$ gcc --version
gcc (Ubuntu 4.4.3-4ubuntu5) 4.4.3

$ cat test.c 
typedef struct Test Test;
typedef struct Test Test;

int main() {}

$ gcc -Wall test.c                                                                                                                                 
test.c:2: error: redefinition of typedef 'Test'
test.c:1: note: previous declaration of 'Test' was here
```